packages/modules-sdk/src/loaders/register-modules.ts

```diff
@@ -0,0 +1,85 @@
+import resolveCwd from "resolve-cwd"
+
+import {
+  MedusaModuleConfig,
+  InternalModuleDeclaration,
+  ModuleDefinition,
+  ModuleResolution,
+  MODULE_SCOPE,
+} from "../types"
+import MODULE_DEFINITIONS from "../definitions"
+
+export const registerModules = ({
+  modules,
+}: MedusaModuleConfig): Record<string, ModuleResolution> => {
+  const moduleResolutions = {} as Record<string, ModuleResolution>
+  const projectModules = modules ?? {}
+
+  for (const definition of MODULE_DEFINITIONS) {
+    const customConfig = projectModules[definition.key]
+    const isObj = typeof customConfig === "object"
+
+    if (isObj && customConfig.scope === MODULE_SCOPE.EXTERNAL) {
+      // TODO: getExternalModuleResolution(...)
+      throw new Error("External Modules are not supported yet.")
+    }
+
+    moduleResolutions[definition.key] = getInternalModuleResolution(
+      definition,
+      projectModules[definition.key] as
+        | InternalModuleDeclaration
+        | false
+        | string
+    )
+  }
+
+  return moduleResolutions
+}
+
+function getInternalModuleResolution(
+  definition: ModuleDefinition,
+  moduleConfig: InternalModuleDeclaration | false | string
+): ModuleResolution {
+  if (typeof moduleConfig === "boolean") {
+    if (!moduleConfig && definition.isRequired) {
+      throw new Error(`Module: ${definition.label} is required`)
+    }
+    if (!moduleConfig) {
+      return {
+        resolutionPath: false,
+        definition,
+        dependencies: [],
+        options: {},
+      }
+    }
+  }
+
+  const isObj = typeof moduleConfig === "object"
```

Comment:
suggestion: Move the is-object util from the core to the utils and use the util everywhere we are doing this operation